### PR TITLE
Add boilerplate routes and controllers

### PR DESCRIPTION
**Pull request by raygdev**
Add controllers directory

ad controllers directory and lettersControllers directory

add lettersControllers.js to set up route controllers for letters routes.

Add lettersRoutes directory

add lettersRoutes directory and lettersRoutes.js
to use lettersControllers

Setup boilerplate for letters controllers

Add comments to controllers

add comments to identify which controller belongs
to which route

Add routes and controllers for letters

### CODE REVIEW

File: server/routes/lettersRoutes/lettersRoutes.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const express = require("express")
+const router = express.Router()
+const lettersControllers = require("../../controllers/lettersControllers/lettersControllers")
+
+router.get("/letters", lettersControllers.getAllLetters)
+router.get("letters/:id", lettersControllers.getLettersById)
```
Comment (raygdev) [inline server/routes/lettersRoutes/lettersRoutes.js:6]: Needs leading forward slash for routes

**Review by raygdev — COMMENTED** [review]
The routes need leading forward slashes. Some were missed

**Review by CampbellSDC — APPROVED** [review]
I think I'm understanding this, and I'm going to work on the routes and controllers tonight - 
Also, can we do a module export in the controllers file, or is the individual exports more performant? Just curious, just trying to learn.

**Comment by raygdev** [comment]
I did it that way to make it explicit which module belongs to what. Also, `exports ` just exports that specific function, and the `module` module exports all of them from that file under the hood as an object. Imagine if we wanted to hook up more routes to our letters routes, instead of immediately exporting, we would have to
```javascript
module.exports = {
   func1,
   func2,
   func3,
 //you get the picture
}
```
Then where we require
```javascript
const { /*multiple named function imports/* } = require("../path/to/file")
```
I think it makes the require look cleaner. But if you have a different suggestion, I'm open to it for sure!

**Pull request merged**